frontend/actions.js

```diff
@@ -95,23 +95,30 @@ export default ({setState, getState}) => {
     }
   }
 
-  const validateSendAddress = (state) => setState({
-    sendAddress: Object.assign({}, state.sendAddress,
-      {validationError: !Cardano.isValidAddress(state.sendAddress.fieldValue) ? {code: 'SendAddressInvalidAddress'} : ''}),
-  })
+  const validateSendAddress = (state) =>
+    setState({
+      sendAddress: Object.assign({}, state.sendAddress, {
+        validationError: !Cardano.isValidAddress(state.sendAddress.fieldValue)
+          ? {code: 'SendAddressInvalidAddress'}
+          : '',
+      }),
+    })
 
   const validateSendAmount = (state) => {
     let validationError = ''
+    const strAmount = state.sendAmount.fieldValue
     const amount = parseFloat(state.sendAmount.fieldValue)
+
     if (isNaN(amount)) {
       validationError = {code: 'SendAmountIsNan'}
     }
-    if (parseFloat(amount.toFixed(6)) <= 0) {
-      validationError = {code: 'SenAmountIsNotPositive'}
+    if (strAmount.split('.').length === 2 && strAmount.split('.')[1].length > 6) {
+      validationError = {code: 'SendAmountPrecisionLimit'}
     }
-    if (parseFloat(amount.toFixed(6)) > state.balance / 1000000) {
-      validationError = {code: 'SendAmountInsufficientFunds', params: {balance: state.balance / 1000000}}
+    if (parseFloat(amount.toFixed(6)) <= 0) {
```

Comment:
Is this different from `amount < 1e-6`?
Also, supplying `0.0000000001` as `strAmount` will overwrite previous `validationError` to something less expected by the user (e.g., `SendAmountIsNotPositive` error seems weird for positive values < 1e-6)

Reply:
fixed